trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js

```diff
@@ -111,25 +113,34 @@ const submitDealAfterUkefIds = async (dealId, dealType, checker) => {
 
     const updatedDeal = await updatedIssuedFacilities(mappedDeal);
 
-    // Update facility stage code to Issued
-    if (mappedDeal.submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.AIN
-      || mappedDeal.submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.MIN
-    ) {
+    /**
+     * Current requirement only allows AIN & MIN deals to be send to ACBS
+     * This call UPDATES facility record by updating their stage from
+     * Unissued (06) to Issued (07)
+     */
+    if (dealController.canDealBeSubmittedToACBS(mappedDeal.submissionType)) {
       await acbsController.issueAcbsFacilities(updatedDeal);
     }
 
     if (shouldUpdateDealFromMIAtoMIN(mappedDeal, tfmDeal)) {
-      const portalMINUpdate = await updatePortalDealFromMIAtoMIN(dealId, checker);
+      const portalMINUpdate = await updatePortalDealFromMIAtoMIN(dealId, dealType, checker);
 
       // NOTE: this is the one and only time that TFM updates a snapshot.
       // Without this, it would involve additional API calls going around in circles.
       const { dealSnapshot } = await api.updateDealSnapshot(dealId, portalMINUpdate);
 
       updatedDeal.submissionType = dealSnapshot.submissionType;
-      updatedDeal.manualInclusionNoticeSubmissionDate = dealSnapshot.details.manualInclusionNoticeSubmissionDate;
-      updatedDeal.checkerMIN = dealSnapshot.details.checkerMIN;
+      if (dealType === CONSTANTS.DEALS.DEAL_TYPE.GEF) {
+        updatedDeal.manualInclusionNoticeSubmissionDate = dealSnapshot.manualInclusionNoticeSubmissionDate;
+        updatedDeal.checkerMIN = dealSnapshot.checkerMIN;
```

Comment:
Are we able to add a test for this? We have one for BSS https://github.com/UK-Export-Finance/dtfs2/blob/main/trade-finance-manager-api/api-tests/v1/deals/deals-submit.second-submission.api-test.js#L303